lawnchair/src/app/lawnchair/ui/preferences/data/liveinfo/model/Announcement.kt

```diff
@@ -0,0 +1,53 @@
+package app.lawnchair.ui.preferences.data.liveinfo.model
+
+import android.util.Log
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toImmutableList
+import org.json.JSONArray
+
+data class Announcement(
+    val text: String,
+    val url: String,
+    val active: Boolean,
+    val test: Boolean,
+) {
+
+    companion object {
+
+        fun fromJsonArray(value: JSONArray): ImmutableList<Announcement> =
+            try {
+                (0 until value.length())
+                    .map { value.getJSONObject(it) }
+                    .map {
+                        Announcement(
+                            text = it.optString("text", ""),
+                            url = it.optString("url", ""),
+                            active = it.optBoolean("active", true),
+                            test = it.optBoolean("test", false),
+                        )
+                    }
+                    .toImmutableList()
+            } catch (e: Exception) {
+                Log.e("Announcement", "fromJsonArray: Failed to parse: ${e.message}")
+                persistentListOf()
+            }
```

Comment:
Same with this. We already have `kotlinx.serialization`, so why not use it

Reply:
Discussing on https://github.com/LawnchairLauncher/lawnchair/pull/3862#discussion_r1429191716